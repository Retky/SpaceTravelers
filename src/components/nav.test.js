
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Nav from './nav';

describe('Nav component', () => {
  test('renders title "Space Travelers\' Hub"', () => {
    render(
      <BrowserRouter><Nav /></BrowserRouter>,
    );

    const title = screen.getByText('Space Travelers\' Hub');
    expect(title).toBeInTheDocument();
  });

  test('component match with the snapshot', () => {
    // Assert
    const tree = renderer.create(<BrowserRouter><Nav /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
