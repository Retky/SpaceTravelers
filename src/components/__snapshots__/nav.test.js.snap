// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component component match with the snapshot 1`] = `
<div>
  <nav>
    <div
      className="logo"
    >
      <img
        alt=""
        src="planet.png"
        width="100%"
      />
    </div>
    <h1>
      Space Travelers' Hub
    </h1>
    <ul
      className="ul"
    >
      <a
        aria-current="page"
        className="link active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "#7fa9fa",
            "textDecoration": "underline",
          }
        }
      >
        <li
          className="rocketslink"
        >
          Rockets
        </li>
      </a>
      <a
        className="link"
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "color": "#1f67f0",
            "textDecoration": "none",
          }
        }
      >
        <li
          className="missionslink"
        >
          Missions
        </li>
      </a>
      <a
        className="link"
        href="/profile"
        onClick={[Function]}
        style={
          Object {
            "color": "#1f67f0",
            "textDecoration": "none",
          }
        }
      >
        <li
          className="profilelink"
        >
          My Profile
        </li>
      </a>
    </ul>
  </nav>
</div>
`;
